flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShim.java

```diff
@@ -205,4 +208,10 @@ SimpleGenericUDAFParameterInfo createUDAFParameterInfo(ObjectInspector[] params,
 	 */
 	void createTableWithConstraints(IMetaStoreClient client, Table table, Configuration conf,
 			UniqueConstraint pk, List<Byte> pkTraits, List<String> notNullCols, List<Byte> nnTraits);
+
+	/**
+	 * Create orc {@link BulkWriter.Factory} for different hive versions.
+	 */
+	BulkWriter.Factory<RowData> createOrcBulkWriterFactory(
```

Comment:
Shouldn't this method be placed in `OrcShim`? It's strange that HiveShim needs to understand how flink-orc works.

Reply:
I think this is hive shim, because orc factory is related to hive version.
If not this, will be a ugly `if else`.

Reply:
I created a orc shim improvement: https://issues.apache.org/jira/browse/FLINK-17785